File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/GooglePlayPublisherPluginExtension.java
package com.jdroid.android.googleplay.publisher;

import com.jdroid.android.googleplay.publisher.commons.PropertyResolver;

import org.gradle.api.Project;

public class GooglePlayPublisherPluginExtension {
	
	private PropertyResolver propertyResolver;
	
	private String applicationId;
	private String privateKeyJsonFilePath;
	
	private String metadataPath;
	private String locales;
	
	private String apkPath;
	private String apkDir;
	private String bundlePath;
	private String bundleDir;

	private String deobfuscationFilePath;
	private Boolean deobfuscationFileUploadEnabled;

	private String track;
	private Double userPercentage;
	private Boolean draft;
	private String releaseName;
	
	private Boolean dryRun;
	
	private Boolean releaseNotesRequired;
	private Boolean videoRequired;
	private Boolean promoGraphicRequired;
	private Boolean phoneScreenshotsRequired;
	private Boolean sevenInchScreenshotsRequired;
	private Boolean tenInchScreenshotsRequired;
	private Boolean wearScreenshotsRequired;
	private Boolean tvScreenshotsRequired;
	private Boolean tvBannerRequired;
	
	private Boolean failOnApkUpgradeVersionConflict;

	private Integer readTimeout;
	private Integer connectTimeout;
	
	public GooglePlayPublisherPluginExtension(Project project) {
		propertyResolver = new PropertyResolver(project);
		
		applicationId = propertyResolver.getStringProp("applicationId");
		privateKeyJsonFilePath = propertyResolver.getStringProp("privateKeyJsonFilePath");
		
		metadataPath = propertyResolver.getStringProp("metadataPath", project.getProjectDir().getAbsolutePath());
		locales = propertyResolver.getStringProp("locales");
		
		apkPath = propertyResolver.getStringProp("apkPath");
		apkDir = propertyResolver.getStringProp("apkDir", project.getProjectDir().getAbsolutePath() + java.io.File.separator + "build" +
				java.io.File.separator + "outputs" + java.io.File.separator + "apk" + java.io.File.separator + "release");
		bundlePath = propertyResolver.getStringProp("bundlePath");
		bundleDir = propertyResolver.getStringProp("bundleDir", project.getProjectDir().getAbsolutePath() + java.io.File.separator + "build" +
						java.io.File.separator + "outputs" + java.io.File.separator + "bundle" + java.io.File.separator + "release");

		deobfuscationFileUploadEnabled = propertyResolver.getBooleanProp("deobfuscationFileUploadEnabled", false);
		deobfuscationFilePath = propertyResolver.getStringProp("deobfuscationFilePath", project.getProjectDir().getAbsolutePath() + java.io.File.separator + "build" +
						java.io.File.separator + "outputs" + java.io.File.separator + "mapping" + java.io.File.separator + "release" + java.io.File.separator + "mapping.txt");

		track = propertyResolver.getStringProp("track");
		userPercentage = propertyResolver.getDoubleProp("userPercentage");
		draft = propertyResolver.getBooleanProp("draft");
		releaseName = propertyResolver.getStringProp("releaseName");
		
		dryRun = propertyResolver.getBooleanProp("dryRun");
		
		releaseNotesRequired = propertyResolver.getBooleanProp("releaseNotesRequired");
		videoRequired = propertyResolver.getBooleanProp("videoRequired");
		promoGraphicRequired = propertyResolver.getBooleanProp("promoGraphicRequired");
		phoneScreenshotsRequired = propertyResolver.getBooleanProp("phoneScreenshotsRequired");
		sevenInchScreenshotsRequired = propertyResolver.getBooleanProp("sevenInchScreenshotsRequired");
		tenInchScreenshotsRequired = propertyResolver.getBooleanProp("tenInchScreenshotsRequired");
		wearScreenshotsRequired = propertyResolver.getBooleanProp("wearScreenshotsRequired");
		tvScreenshotsRequired = propertyResolver.getBooleanProp("tvScreenshotsRequired");
		tvBannerRequired = propertyResolver.getBooleanProp("tvBannerRequired");
		
		failOnApkUpgradeVersionConflict = propertyResolver.getBooleanProp("failOnApkUpgradeVersionConflict");

		readTimeout = propertyResolver.getIntegerProp("readTimeout");
		connectTimeout = propertyResolver.getIntegerProp("connectTimeout");
	}
	
	public void setApplicationId(String applicationId) {
		this.applicationId = applicationId;
	}
	
	public String getApplicationId() {
		return applicationId;
	}
	
	public String getPrivateKeyJsonFilePath() {
		return privateKeyJsonFilePath;
	}
	
	public void setPrivateKeyJsonFilePath(String privateKeyJsonFilePath) {
		this.privateKeyJsonFilePath = privateKeyJsonFilePath;
	}
	
	public String getMetadataPath() {
		return metadataPath;
	}
	
	public void setMetadataPath(String metadataPath) {
		this.metadataPath = metadataPath;
	}
	
	public String getLocales() {
		return locales;
	}
	
	public void setLocales(String locales) {
		this.locales = locales;
	}
	
	public String getApkPath() {
		return apkPath;
	}
	
	public void setApkPath(String apkPath) {
		this.apkPath = apkPath;
	}
	
	public String getBundlePath() {
		return bundlePath;
	}
	
	public void setBundlePath(String bundlePath) {
		this.bundlePath = bundlePath;
	}
	
	public String getTrack() {
		return track;
	}
	
	public void setTrack(String track) {
		this.track = track;
	}
	
	public Boolean getDraft() {
		return draft;
	}
	
	public void setDraft(Boolean draft) {
		this.draft = draft;
	}
	
	public String getReleaseName() {
		return releaseName;
	}
	
	public void setReleaseName(String releaseName) {
		this.releaseName = releaseName;
	}
	
	public Boolean getReleaseNotesRequired() {
		return releaseNotesRequired;
	}
	
	public void setReleaseNotesRequired(Boolean releaseNotesRequired) {
		this.releaseNotesRequired = releaseNotesRequired;
	}
	
	public Boolean getVideoRequired() {
		return videoRequired;
	}
	
	public void setVideoRequired(Boolean videoRequired) {
		this.videoRequired = videoRequired;
	}
	
	public Boolean getPromoGraphicRequired() {
		return promoGraphicRequired;
	}
	
	public void setPromoGraphicRequired(Boolean promoGraphicRequired) {
		this.promoGraphicRequired = promoGraphicRequired;
	}
	
	public Boolean getPhoneScreenshotsRequired() {
		return phoneScreenshotsRequired;
	}
	
	public void setPhoneScreenshotsRequired(Boolean phoneScreenshotsRequired) {
		this.phoneScreenshotsRequired = phoneScreenshotsRequired;
	}
	
	public Boolean getSevenInchScreenshotsRequired() {
		return sevenInchScreenshotsRequired;
	}
	
	public void setSevenInchScreenshotsRequired(Boolean sevenInchScreenshotsRequired) {
		this.sevenInchScreenshotsRequired = sevenInchScreenshotsRequired;
	}
	
	public Boolean getTenInchScreenshotsRequired() {
		return tenInchScreenshotsRequired;
	}
	
	public void setTenInchScreenshotsRequired(Boolean tenInchScreenshotsRequired) {
		this.tenInchScreenshotsRequired = tenInchScreenshotsRequired;
	}
	
	public Boolean getWearScreenshotsRequired() {
		return wearScreenshotsRequired;
	}
	
	public void setWearScreenshotsRequired(Boolean wearScreenshotsRequired) {
		this.wearScreenshotsRequired = wearScreenshotsRequired;
	}
	
	public Boolean getTvScreenshotsRequired() {
		return tvScreenshotsRequired;
	}
	
	public void setTvScreenshotsRequired(Boolean tvScreenshotsRequired) {
		this.tvScreenshotsRequired = tvScreenshotsRequired;
	}
	
	public Boolean getTvBannerRequired() {
		return tvBannerRequired;
	}
	
	public void setTvBannerRequired(Boolean tvBannerRequired) {
		this.tvBannerRequired = tvBannerRequired;
	}
	
	public Boolean getFailOnApkUpgradeVersionConflict() {
		return failOnApkUpgradeVersionConflict;
	}
	
	public void setFailOnApkUpgradeVersionConflict(Boolean failOnApkUpgradeVersionConflict) {
		this.failOnApkUpgradeVersionConflict = failOnApkUpgradeVersionConflict;
	}
	
	public String getApkDir() {
		return apkDir;
	}
	
	public void setApkDir(String apkDir) {
		this.apkDir = apkDir;
	}
	
	public String getBundleDir() {
		return bundleDir;
	}
	
	public void setBundleDir(String bundleDir) {
		this.bundleDir = bundleDir;
	}
	
	public Double getUserPercentage() {
		return userPercentage;
	}
	
	public void setUserPercentage(Double userPercentage) {
		this.userPercentage = userPercentage;
	}
	
	public Boolean getDryRun() {
		return dryRun;
	}
	
	public void setDryRun(Boolean dryRun) {
		this.dryRun = dryRun;
	}

	public Boolean isDeobfuscationFileUploadEnabled() {
		return deobfuscationFileUploadEnabled;
	}

	public void setDeobfuscationFileUploadEnabled(Boolean deobfuscationFileUploadEnabled) {
		this.deobfuscationFileUploadEnabled = deobfuscationFileUploadEnabled;
	}

	public String getDeobfuscationFilePath() {
		return deobfuscationFilePath;
	}

	public void setDeobfuscationFilePath(String deobfuscationFilePath) {
		this.deobfuscationFilePath = deobfuscationFilePath;
	}
}

File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/GooglePlayPublisherPluginExtension.kt
package com.jdroid.android.googleplay.publisher

import com.jdroid.android.googleplay.publisher.commons.PropertyResolver
import org.gradle.api.Project
import java.io.File

open class GooglePlayPublisherPluginExtension(project: Project) {

    private val propertyResolver: PropertyResolver = PropertyResolver(project)

    var applicationId: String? = propertyResolver.getStringProp("applicationId")
    var privateKeyJsonFilePath: String? = propertyResolver.getStringProp("privateKeyJsonFilePath")
    var metadataPath: String? = propertyResolver.getStringProp("metadataPath", project.projectDir.absolutePath)
    var locales: String? = propertyResolver.getStringProp("locales")
    var apkPath: String? = propertyResolver.getStringProp("apkPath")
    var apkDir: String? = propertyResolver.getStringProp(
        "apkDir", project.projectDir.absolutePath + File.separator + "build" +
            File.separator + "outputs" + File.separator + "apk" + File.separator + "release"
    )
    var bundlePath: String? = propertyResolver.getStringProp("bundlePath")
    var bundleDir: String? = propertyResolver.getStringProp(
        "bundleDir", project.projectDir.absolutePath + File.separator + "build" +
            File.separator + "outputs" + File.separator + "bundle" + File.separator + "release"
    )
    var deobfuscationFilePath: String? = propertyResolver.getStringProp(
        "deobfuscationFilePath", project.projectDir.absolutePath + File.separator + "build" +
            File.separator + "outputs" + File.separator + "mapping" + File.separator + "release" + File.separator + "mapping.txt"
    )
    var isDeobfuscationFileUploadEnabled: Boolean? = propertyResolver.getBooleanProp("deobfuscationFileUploadEnabled", false)
    var track: String? = propertyResolver.getStringProp("track")
    var userPercentage: Double? = propertyResolver.getDoubleProp("userPercentage")
    var draft: Boolean? = propertyResolver.getBooleanProp("draft")
    var releaseName: String? = propertyResolver.getStringProp("releaseName")
    var dryRun: Boolean? = propertyResolver.getBooleanProp("dryRun")
    var releaseNotesRequired: Boolean? = propertyResolver.getBooleanProp("releaseNotesRequired")
    var videoRequired: Boolean? = propertyResolver.getBooleanProp("videoRequired")
    var promoGraphicRequired: Boolean? = propertyResolver.getBooleanProp("promoGraphicRequired")
    var phoneScreenshotsRequired: Boolean? = propertyResolver.getBooleanProp("phoneScreenshotsRequired")
    var sevenInchScreenshotsRequired: Boolean? = propertyResolver.getBooleanProp("sevenInchScreenshotsRequired")
    var tenInchScreenshotsRequired: Boolean? = propertyResolver.getBooleanProp("tenInchScreenshotsRequired")
    var wearScreenshotsRequired: Boolean? = propertyResolver.getBooleanProp("wearScreenshotsRequired")
    var tvScreenshotsRequired: Boolean? = propertyResolver.getBooleanProp("tvScreenshotsRequired")
    var tvBannerRequired: Boolean? = propertyResolver.getBooleanProp("tvBannerRequired")
    var failOnApkUpgradeVersionConflict: Boolean? = propertyResolver.getBooleanProp("failOnApkUpgradeVersionConflict")
    var readTimeout: Int? = propertyResolver.getIntegerProp("readTimeout")
    var connectTimeout: Int? = propertyResolver.getIntegerProp("connectTimeout")
}
